modules/core/src/v2/coins/trx.ts

```diff
@@ -249,6 +275,200 @@ export class Trx extends BaseCoin {
+    return hdNode.keyPair.__Q.getEncoded(false).toString('hex');
+  }
+
+  compressedPubToHexAddress(pub: string): string {
+    const byteArrayAddr = bitgoAccountLib.Trx.Utils.getByteArrayFromHexAddress(pub);
+    const rawAddress = bitgoAccountLib.Trx.Utils.getRawAddressFromPubKey(byteArrayAddr);
+    return Buffer.from(rawAddress).toString('hex').toUpperCase();
+  }
+
+  xprvToCompressedPrv(xprv: string): string {
+    if (!this.isValidXprv(xprv)) {
+      throw new Error('invalid xprv');
+    }
+
+    const hdNode = utxoLib.HDNode.fromBase58(xprv, this.bitcoinEncoding());
+    return hdNode.keyPair.d.toBuffer(32).toString('hex');
+  };
+
+  bitcoinEncoding(): any {
+    return {
+      messagePrefix: '\x18Bitcoin Signed Message:\n',
+      bech32: 'bc',
+      bip32: {
+        public: 0x0488b21e,
+        private: 0x0488ade4
+      },
+      pubKeyHash: 0x00,
+      scriptHash: 0x05,
+      wif: 0x80,
+      coin: 'btc',
+    };
+  }
+
+  /**
+   * Make a query to Trongrid for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @param callback
+   * @returns {Object} response from Trongrid
+   */
+  recoveryPost(query: { path: string, jsonObj: any, node: NodeTypes }, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co(function*() {
+      let nodeUri = '';
+      switch (query.node) {
+        case NodeTypes.Full:
+          nodeUri = common.Environments[self.bitgo.getEnv()].tronNodes.full;
+          break;
+        case NodeTypes.Solidity:
+          nodeUri = common.Environments[self.bitgo.getEnv()].tronNodes.solidity;
+          break;
+        default:
+          throw new Error('node type not found');
+      }
+
+      const response = yield request
+          .post(nodeUri + query.path)
+          .send(query.jsonObj);
+
+      if (!response.ok) {
+        throw new Error('could not reach Tron node');
+      }
+      return response.body;
+    })
+        .call(this)
+        .asCallback(callback);
+  }
+
+  /**
+   * Query our explorer for the balance of an address
+   * @param address {String} the address encoded in hex
+   * @param callback
+   * @returns {BigNumber} address balance
+   */
+  getAccountBalanceFromNode(address: string, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co(function*() {
+      const result = yield self.recoveryPost({
+        path: '/walletsolidity/getaccount',
+        jsonObj: { address },
+        node: NodeTypes.Solidity,
+      });
+      return result.balance;
+    })
+      .call(this)
+      .asCallback(callback);
+  }
+
+  /**
+   * Retrieves our build transaction from a node.
+   * @param toAddr hex-encoded address
+   * @param fromAddr hex-encoded address
+   * @param amount
+   * @param callback
+   */
+  getBuildTransaction(toAddr: string, fromAddr: string, amount: number, callback?: NodeCallback<any>): Bluebird<any> {
```

Comment:
Make private if not meant for external usage. Also can we use a more specific return type?